### ISSUE TITLE
Discuss the need for placing annotations in separate package

### ISSUE THREAD

**Issue by chkal** [mvc-spec/mvc-spec#159]
I'm currently reviewing the classes and API docs and discovered something I would like to discuss.

We are currently inconsistent about the packages used for annotations. There are basically two different ways it is done today:

* Most annotations are placed in a separate package `javax.mvc.annotation`.
* In one case an annotation is placed in the package named after the purpose. So `@MvcBinding` is located in the `javax.mvc.binding` package.

I don't like this inconsistency. We should choose where to place annotations and be consistent about it. Either we place them in one common package or in the package containing class/interfaces for this specific "feature" of the spec (binding/security/etc).

To be honest, I like the latter one. IMO there is no real benefit of placing all annotations in a single package. This looks like an anti-pattern to me. Somehow similar to package-by-layer instead of package-by-module.

I just checked the Java EE API docs. I don't know how many specs are contained in Java EE (perhaps about 20?), but there are only 2 specs with a separate annotations package (Servlet and JAXB). All other specs (JAX-RS, CDI, JPA, EJB, etc) place annotations in regular packages together with other types.

So what do you think? Should we remove the `javax.mvc.annotation` package and move the annotations to the matching packages?

I know that this is a pretty big change. But it could be worth the effort. Today we can move types into packages which are a better fit. But after releasing the spec, we can NEVER change it again.

**Comment by lefloh** [comment]
+1 for moving the annotations to the packages where they belong.

**Comment by chkal** [comment]
Ok, so **IF** we decide to move the annotations, we could handle it like this:

Currently there are 5 classes in the `annotation` package:

```
javax.mvc.annotation.Controller
javax.mvc.annotation.View
javax.mvc.annotation.RedirectScoped
javax.mvc.annotation.UriRef
javax.mvc.annotation.CsrfValid
```

My idea would be to move them like this:

```
javax.mvc.Controller
javax.mvc.View
javax.mvc.RedirectScoped
javax.mvc.UriRef
javax.mvc.security.CsrfValid
```

But I'm not sure if we should really move `@UriRef` to the root package. In this case we would have
two classes related to URI building in the root package: `@UriRef` and `MvcUriBuilder`. So perhaps
we should create a new `javax.mvc.uri` package, ending up with:

```
javax.mvc.Controller
javax.mvc.View
javax.mvc.RedirectScoped
javax.mvc.uri.UriRef
javax.mvc.security.CsrfValid
```

Thoughts?

**Comment by Daniel-Dos** [comment]
show. 

I like the new nomenclature of packages.

**Comment by lefloh** [comment]
I also think an `uri` package makes sense.
Unsure if `@View` could also be moved to `javax.mvc.engine`?

**Comment by chkal** [comment]
> Unsure if @View could also be moved to javax.mvc.engine?

I also thought about that. But actually the `engine` package contains (as the name suggests) classes for the `ViewEngine` SPI, and `@View` is a more general annotation. I guess it is important enough to put it into the root package?

**Comment by lefloh** [comment]
I'm not strong on this one. The root package is fine.

**Comment by chkal** [comment]
Thanks for all your feedback. I'll create a corresponding pull request later today or tomorrow.

**Comment by chkal** [comment]
I just created #163 to apply the discussed changes.

**Comment by chkal** [comment]
I just merged #163. So this issue is resolved.
